variants/DISCO_F413ZH/variant.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2019, STMicroelectronics
+ * All rights reserved.
+ *
+ * This software component is licensed by ST under BSD 3-Clause license,
+ * the "License"; You may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *                        opensource.org/licenses/BSD-3-Clause
+ *
+ *******************************************************************************
+ */
+
+#include "pins_arduino.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Digital PinName array
+// This array allows to wrap Arduino pin number(Dx or x)
+// to STM32 PinName (PX_n)
+const PinName digitalPin[] = {
+  PF_6,  // D0 - UART7 RX
+  PF_7,  // D1 - UART7 TX
+  PG_13, // D2
+  PF_10, // D3
+  PB_6,  // D4
+  PE_6,  // D5
+  PB_0,  // D6
+  PC_13, // D7
+  PA_4,  // D8
+  PB_8,  // D9
+  PA_15, // D10 - SPI_SSEL
+  PB_5,  // D11 - SPI_MOSI
+  PB_4,  // D12 - SPI_MISO
+  PB_12, // D13 - SPI_SCK, LED_GREEN
+  PB_11, // D14 - I2C_SDA
+  PB_10, // D15 - I2C_SCL
+  PC_0,  // D16/A0
+  PA_1,  // D17/A1
+  PA_2,  // D18/A2
+  PA_5,  // D19/A3
+  PB_1,  // D20/A4
+  PC_4,  // D21/A5
+  // not on connector
+  PG_14, // D22 - STLink Tx
+  PG_9,  // D23 - STLink Rx
+  PE_3,  // D24 - LD4, LED1_RED
+  PC_5,  // D25 - LD3, LED2_GREEN
+  PA_0,  // D26 - User button
+  PB_4,  // D27 - ISM43362_WIFI_MISO
+  PB_5,  // D28 - ISM43362_WIFI_MOSI
+  PB_12, // D29 - ISM43362_WIFI_SCLK
+  PG_11, // D30 - ISM43362_WIFI_NSS
+  PH_1,  // D31 - ISM43362_WIFI_RESET
+  PG_12, // D32 - ISM43362_WIFI_DATAREADY
+  PB_15  // D33 - ISM43362_WIFI_WAKEUP
+};
+
+#ifdef __cplusplus
+}
+#endif
+
+// ----------------------------------------------------------------------------
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+  * @brief  System Clock Configuration
+  * @param  None
+  * @retval None
+  */
+WEAK void SystemClock_Config(void)
+{
+  /* In DISCO_F413ZH board, Arduino connector and Wifi embeded module are sharing the same SPI pins */
+  /* We need to set the default SPI SS pin for the Wifi module to the inactive state i.e. 1 */
+  /* See board User Manual: WIFI_SPI_CS = PG_11*/
+  __HAL_RCC_GPIOG_CLK_ENABLE();
+  GPIO_InitTypeDef  GPIO_InitStruct;
+  GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+  GPIO_InitStruct.Pull = GPIO_NOPULL;
+  GPIO_InitStruct.Speed = GPIO_SPEED_FAST;
+  GPIO_InitStruct.Pin = GPIO_PIN_11;
+  HAL_GPIO_Init(GPIOG, &GPIO_InitStruct);
+  HAL_GPIO_WritePin(GPIOG, GPIO_PIN_11, GPIO_PIN_SET);
```

Comment:
GPIO configuration has nothing to do with clock configuration. It should be moved in another function (to be created in variant.cpp) `void initVariant(void)`

Reply:
I agree, but it was placed in [`SystemInit()` of  `mbed-os/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F413xH/TARGET_DISCO_F413ZH/system_clock.c`](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F413xH/TARGET_DISCO_F413ZH/system_clock.c#L100).  Moving this code to freshly created `void initVariant()` does not allow Arduino SPI and Wi-FI coexist, SD-card shield does not work. Do you have other ideas?

Reply:
SystemInit() is more generic than SystemClock_Config(), and we don't have SystemInit(). Whatever it is mbed concerns.
About `void initVariant()`  @fpistm fixes an issue yesterday about that: #993
It is now merged in master branch.
Can you test with this patch ?

Reply:
This method works. Will include it into the next commits.